packages/rrweb/src/record/observers/canvas/canvas-manager.ts

```diff
@@ -210,15 +341,32 @@ export class CanvasManager implements CanvasManagerInterface {
     let lastSnapshotTime = 0;
     let rafId: number;
 
-    const getCanvas = (): HTMLCanvasElement[] => {
+    const getCanvas = (
+      canvasElement?: HTMLCanvasElement,
+    ): HTMLCanvasElement[] => {
       const matchedCanvas: HTMLCanvasElement[] = [];
-      win.document.querySelectorAll('canvas').forEach((canvas) => {
+      if (canvasElement) {
         if (
-          !isBlocked(canvas, blockClass, blockSelector, unblockSelector, true)
+          !isBlocked(
+            canvasElement,
+            blockClass,
+            blockSelector,
+            unblockSelector,
+            true,
+          )
         ) {
-          matchedCanvas.push(canvas);
+          matchedCanvas.push(canvasElement);
         }
-      });
+      } else {
```

Comment:
I would change up the structure a bit and make use of an early `return` to make the diff a bit easier to read (and simplifies the if/else):

```
if (canvasElement) {
  return [canvasElement]
}

// ... the old code here ...
```